feg/gateway/services/s6a_proxy/servicers/ul.go

```diff
@@ -158,3 +189,17 @@ func (s *s6aProxy) UpdateLocationImpl(req *protos.UpdateLocationRequest) (*proto
 	}
 	return res, err
 }
+
+// getFeatureListID2 creates the Feature-List-ID 2 based on TS 29.272
+func getFeatureListID2(supportedFeatures []SupportedFeatures) *protos.FeatureListId2 {
+	protoFeatureList := &protos.FeatureListId2{}
+	for _, features := range supportedFeatures {
+		if features.FeatureListID == 2 {
+			// get bit 27 (TS 29.272 Nr As Secondary Rat)
+			if features.FeatureList&(1<<27) != 0 {
+				protoFeatureList.NrAsSecondaryRat = true
```

Comment:
return &protos.FeatureListId2{NrAsSecondaryRat: true}

Reply:
My idea was this code to be easily extensible once we have more flags. There are 28 other potential flags we will be using, specially with 5g.

If I change the code to return in case of finding bit 27, the day we add a new flag, we will have to modify the algorithm back to its current implementation.

Also, I thought it was better to return an empty slice instead of a nil to avoid nil pointer panics?